uiowa/uiowa
# Guest hawkid support (in uiowa_auth)

https://github.com/uiowa/uiowa_auth/pull/8
Resolves https://github.com/uiowa/uiowa/issues/6101

Guest hawkids don't usually have member/group information. Existing functionality assumed it would always return an array. This PR by way of https://github.com/uiowa/uiowa_auth add is_array() checks around multiple points in the code for user sync, login, and linking. This will maintain the checks against current roles and whether other roles should be applied (e.g. Web team being assigned the admin role).

# How to test

- This has been pushed out to DEV for testing on default.dev.drupal.uiowa.edu.
- Note that guest hawkid [dworrll](https://default.dev.drupal.uiowa.edu/user/61) was able to login and has the editor role only which was added on account creation.
- Login as user 1 and remove roles from your hawkid. `ddev drush @default.dev uli /admin/people`
- In an anonymous browser session, login with your hawkid. Web team should get administrator role applied. Users like Michael and Megan should get the webmaster role. Mara (doesn't have an account on default.dev nor role mappings to admin/webmaster) should not be able to login.
- Not sure how it was tested originally, but ideally, if someone loses the necessary group, their auto-assigned role should be revoked. We could draw straws to temporarily drop someone via https://iam.uiowa.edu/access/

# Post Deploy

- Follow up with paklee email/ticket.
